front/app/modules/commercial/moderation/admin/containers/ModerationRow.tsx

```diff
@@ -168,6 +153,34 @@ const ModerationRow = memo<Props & InjectedIntlProps>(
       win && win.focus();
     };
 
+    function getViewLink(moderatableType: TModeratableType) {
+      if (moderatableType === 'Comment') {
+        if (
+          belongsToTypes.includes('initiative') &&
+          moderation.attributes.belongs_to.initiative?.slug
+        ) {
+          return `/initiatives/${moderation.attributes.belongs_to.initiative.slug}`;
+        }
+
+        if (
+          belongsToTypes.includes('idea') &&
+          moderation.attributes.belongs_to.idea?.slug
+        ) {
+          return `/ideas/${moderation.attributes.belongs_to.idea.slug}`;
+        }
+      }
+
+      if (moderatableType === 'Idea') {
+        return `/ideas/${moderation.attributes.content_slug}`;
+      }
+
+      if (moderatableType === 'Initiative') {
+        return `/initiatives/${moderation.attributes.content_slug}`;
+      }
+
+      return null;
+    }
+
     return (
```

Comment:
100 lines is still fine 😎 I don't see the immediate benefit of extracting here. It's one component that belongs together and it's not confusing because the code outside of the return is limited. The benefit we currently have is that you can see everything that is one component at one glance and edit right away. 

Not saying more extraction would be wrong, but not worth it at the moment imo.